internal/bminventory/inventory.go

```diff
@@ -5278,12 +5274,12 @@ func (b *bareMetalInventory) V2UpdateHostInstallProgressInternal(ctx context.Con
 		if stageChanged {
 			if err := b.clusterApi.UpdateInstallProgress(ctx, *host.ClusterID); err != nil {
 				log.WithError(err).Errorf("failed to update cluster %s progress", host.ClusterID)
-				return host, err
+				return common.NewApiError(http.StatusInternalServerError, err)
```

Comment:
I think for something like this you can just return the error because we're using `GenerateErrorResponder` in the caller.

Reply:
Gotcha, modified as such! Thanks!